File: Movie4U/Models/GenreModel.cs
using Movie4U.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Movie4U.Models
{
    public class GenreModel
    {
        public string genre { get; set; }

        [Required, Key]
        public int genre_id { get; set; }
    
        public void Copy(Genre source)
        {
            this.genre = source.genre;
            this.genre_id = source.genre_id;
        }

    }
}

File: flutter_app/lib/models/genre_model.dart
class GenreModel {
  GenreModel({
    required this.genre,
    required this.genreId,
  });

  String genre;
  int genreId;

  factory GenreModel.fromJson(Map<String, dynamic> json) => GenreModel(
        genre: json["genre"],
        genreId: json["genre_id"],
      );

  Map<String, dynamic> toJson() => {
        "genre": genre,
        "genre_id": genreId,
      };
}
